File: 3sem/4lab/Banks.Console/Handlers/ExitHandler.cs
using Banks.Services;

namespace Banks.Console.Handlers;

public class ExitHandler : Handler
{
    public override void Handle(string request, CentralBank centralBank)
    {
        ArgumentNullException.ThrowIfNull(request);
        ArgumentNullException.ThrowIfNull(centralBank);

        if (request.ToLower() == "exit")
        {
            System.Environment.Exit(0);
        }

        NextHandler?.Handle(request, centralBank);
    }
}

File: 4sem/1lab/src/main/java/BanksRemake/Presentation/Handlers/ExitHandler.java
package BanksRemake.Presentation.Handlers;

import BanksRemake.Domain.Exceptions.InvalidBankAccountOperationException;
import BanksRemake.Domain.Exceptions.InvalidBankOperationException;
import BanksRemake.Domain.Exceptions.InvalidDepositInterestRateSelectorException;
import BanksRemake.Domain.Exceptions.InvalidMoneyGapException;
import BanksRemake.Domain.Services.CentralBank;

import java.util.Scanner;

public class ExitHandler extends HandlerBase {
    @Override
    public void handle(String request, CentralBank centralBank, Scanner scanner) throws InvalidBankOperationException, InvalidBankAccountOperationException, InvalidDepositInterestRateSelectorException, InvalidMoneyGapException {
        if (request.toLowerCase().equals("exit")) {
            scanner.close();
            System.exit(0);
        }

        if (getNext() != null) {
            getNext().handle(request, centralBank, scanner);
        }
    }
}
